go/libraries/doltcore/doltdb/durable/index.go

```diff
@@ -40,53 +83,69 @@ func NewIndexSet(ctx context.Context, vrw types.ValueReadWriter) IndexSet {
 	}
 }
 
-func MapFromIndexSet(ic IndexSet) types.Map {
-	return ic.(nomsIndexSet).indexes
-}
-
 type nomsIndexSet struct {
 	indexes types.Map
 	vrw     types.ValueReadWriter
 }
 
 var _ IndexSet = nomsIndexSet{}
 
-func (c nomsIndexSet) GetIndex(ctx context.Context, name string) (types.Map, error) {
-	v, ok, err := c.indexes.MaybeGet(ctx, types.String(name))
+// HashOf implementes IndexSet.
+func (s nomsIndexSet) HashOf() (hash.Hash, error) {
+	return s.indexes.Hash(s.vrw.Format())
+}
+
+// GetIndex implementes IndexSet.
```

Comment:
Do you have a generator for this or something

Reply:
multi-cursor!